torch/csrc/autograd/engine.cpp

```diff
@@ -368,6 +369,17 @@ void GraphTaskGuard::restore_current_graph_task() {
   current_graph_task = std::move(last_graph_task_);
 }
 
+// The current graph task's exec_info could be used to trim unnecessary edegs
+// during node evaluation, see `Node.task_should_compute_output()` function.
+const std::unordered_map<Node*, GraphTask::ExecInfo>*
+get_current_graph_task_exec_info() {
+  return current_graph_task ? &current_graph_task->exec_info_ : nullptr;
```

Comment:
I think this should assert that there is a current graph_task?
Calling `task_should_compute_output()` outside of the backward pass doesn't make sense.

Reply:
https://github.com/pytorch/pytorch/blob/c54d18dbc7bb2f9fdd83c5de529702e5a02295c3/test/test_autograd.py#L641-L650

@albanD this test will fail if we add `TORCH_INTERNAL_ASSERT(current_graph_task);`, 

https://github.com/pytorch/pytorch/blob/c54d18dbc7bb2f9fdd83c5de529702e5a02295c3/test/test_autograd.py#L650

Specifically, it is this line that failed, because we are running backward function without a `current_graph_task`

Reply:
I believe this is the last thread we need to resolve.

Reply:
Sounds like we should just compute all outputs in that case

Reply:
Yep, I retract my statement that it doesn't make sense I guess then. Assuming that everything is needed sounds good.

Reply:
Yes, in the current code, if current_graph_task is not defined, then it will compute everything, which is expected then.
Do you guys have any other comments for this PR?